### ISSUE TITLE
DNF occasionally saved as Null

### ISSUE THREAD

**Issue by ThisTemba** [ThisTemba/cubing-website#26]
DNF solves should be saved in Firebase as the number `Infinity`. Sometimes they get saved as `null`. This affects the session's best single.

**Comment by ThisTemba** [comment]
### Cause:
* JSON can't store `Infinity`. See [this Stack Overflow post]( https://stackoverflow.com/questions/1423081/json-left-out-infinity-and-nan-json-status-in-ecmascript)
* localStorage uses JSON to store data
* Sessions are stored in local storage on `timePage.jsx`:
    ```
    const [session, setSession] = useLocalStorage("session", {
        name: null,
        solves: [],
      });
    ```

### Steps to Recreate:
1. Start a new session
2. Create a dummy solve and mark it as "DNF"
3. Reload the page

You should see that instead of saying "DNF" the solve now has a time of `0.00`. Additionally, if you check your browser's local storage the `dur` property of the solve will be `null`

**Comment by ThisTemba** [comment]
### Possible Solutions:
* Don't store sessions in local storage (store in firebase instead)
* Don't use `Infinity` for "DNF"
* Parse sessions from local storage data after read

**Comment by ThisTemba** [comment]
### Evaluating Solutions:
1. Storing sessions in Firebase is the ideal long-term solution as this would allow users to pick up where they left off on different devices, but implementing that is a lot of work. It is hard partly because editing data models is laborious in pretty much any situation, but also because if we simply replaced local storage with firebase, we would be writing to firebase after every solve. This is more writes than necessary. The ideal here would be writing to firebase every 5 minutes or so, or having a button that says "save session". 

2. Using `infinity` for "DNF" is quite useful[^1]  so we can eliminate option 2.

3. That leave the third option which is parsing sessions from local storage. This fix could be as simple as 
```javascript
if (localStorageDur === null) dur = Infinity
```



[^1]: I wrote about this here: [⌛ Duration is a number](https://tembamateke.notion.site/Duration-is-a-number-40284d58e612431fb472422f97fc8a45)

**Comment by ThisTemba** [comment]
Fixed with #27
